sim/midas/src/main/cc/bridges/test_finisher.cc

```diff
@@ -0,0 +1,37 @@
+#ifdef TESTFINISHERBRIDGEMODULE_struct_guard
+
+#include "test_finisher.h"
+#include <iostream>
+
+test_finisher_t::test_finisher_t(
+	simif_t* sim, 
+  TESTFINISHERBRIDGEMODULE_struct * mmio_addrs,
+	unsigned int num_messages,
+  unsigned int* isErr,
+	const char* const* msgs): 
+	bridge_driver_t(sim), 
+	mmio_addrs(mmio_addrs), 
+	isErr(isErr),
+	msgs(msgs)
+{
+  this->mmio_addrs = mmio_addrs;
+}
+
+test_finisher_t::~test_finisher_t() {
+	free(this->mmio_addrs);
+}
+
+void test_finisher_t::tick() {  //reads the MMIOs every tick
```

Comment:
Consider not polling on every tick invocation. This should eventually be on a +arg, but for now start by having some constant value called out in the header. Polling every 10 iterations will probably help improve FMR without meaningfully extending metasimulation runtimes.